### PR TITLE
greenlight: allow for provider-type in user:create

### LINKED ISSUE #135: Expand "create Greenlight users" to allow setting the provider
Hi,

https://github.com/ebbba-org/ansible-role-bigbluebutton/blob/a17a01393cfec76c52d6b62f18f84d1e5bdca935/tasks/greenlight.yml#L50

doesn't allow setting the `provider` field listed in `rake --tasks`-output:
```
rake user:create[name,email,password,role,provider]  # Creates a user account
```
I'd like to create an LDAP user account with "admin" role so I don't need to `UPDATE` something in Postgres after the role is done...

### PR DESCRIPTION
**Pull request by einhirn**
Fixes #135 - although I'm not sure if it's actually useful, because it still waits for changes in Greenlight...

Perhaps it helps with other providers than LDAP? Or other use cases?

### CODE REVIEW

**Review by Davka — APPROVED** [review]

**Review by n0emis — CHANGES_REQUESTED** [review]
This PR looks good so far, although some documentation of this feature in the README.md would be preferable...

**Review by BrutalBirdie — CHANGES_REQUESTED** [review]
README still missing

**Comment by einhirn** [comment]
As "README.md" says see `defaults/main.yml` for details, I've added some explanation there.

This can be merged, but might not work as expected yet. There needs to be another PR once Greenlight accepts a PR TBD for updating `rake user:add` etc. It currently misses the ability to specify the "social_id" like "uid=X,ou=Y,dc=Z" for LDAP or whatever else for other auth plugins.

**Comment by crosscodr** [comment]
Hi @einhirn, this looks good, but was this implemented in greenlight yet?

**Comment by einhirn** [comment]
Hmm... Too long ago - I think this works only partially, because Ahmad Farhat of Greenlight just said something along the lines "rake:user was only ever meant to create local accounts". https://github.com/bigbluebutton/greenlight/issues/2742

Now it comes back to me: The new field "social_id" can't be set currently, so I can't add an ldap user as admin like i wanted...

I guess I'll create a PR for Greenlight in about 3 weeks from now... Didn't get around to do that yet.

**Comment by crosscodr** [comment]
On my greenlight 2.9.2 instance rake -T shows
`rake user:create[name,email,password,role,provider]  # Creates a user account`
So I think this PR would work now.

**Review by crosscodr — APPROVED** [review]

File: tasks/greenlight.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
     # https://docs.bigbluebutton.org/greenlight/gl-admin.html#creating-accounts
     - name: Create greenlight users
-      command: "docker exec greenlight-v2 bundle exec rake user:create[\"{{ item.name }}\",\"{{ item.email }}\",\"{{ item.password }}\",\"{{ item.type }}\"]"
+      command: "docker exec greenlight-v2 bundle exec rake user:create[\"{{ item.name }}\",\"{{ item.email }}\",\"{{ item.password }}\",\"{{ item.type }}{{ ',' + item.provider if 'provider' in item.keys() else '' }}\"]"
```
Comment (crosscodr) [inline tasks/greenlight.yml:42]:
```suggestion
      command: "docker exec greenlight-v2 bundle exec rake user:create[\"{{ item.name }}\",\"{{ item.email }}\",\"{{ item.password }}\",\"{{ item.type }}\"{{ ',' + '"' + item.provider + '"' if 'provider' in item.keys() else '' }}]"
```

**Review by crosscodr — CHANGES_REQUESTED** [review]

**Review by Davka — APPROVED** [review]

**Comment by einhirn** [comment]
> 
> 
> On my greenlight 2.9.2 instance rake -T shows `rake user:create[name,email,password,role,provider] # Creates a user account` So I think this PR would work now.

No, "social_id" is still missing, and I didn't get around to prepare a PR against Greenlight yet... Maybe just close this PR? I don't know when I'll be able to get to it - also when a new PR would be merged to Greenlight upstream.

**Comment by crosscodr** [comment]
> > On my greenlight 2.9.2 instance rake -T shows `rake user:create[name,email,password,role,provider] # Creates a user account` So I think this PR would work now.
> 
> No, "social_id" is still missing, and I didn't get around to prepare a PR against Greenlight yet... Maybe just close this PR? I don't know when I'll be able to get to it - also when a new PR would be merged to Greenlight upstream.

Yeah, thats right :(
I think we can still leave this PR as it is, in case this gets into greenlight some times.

**Comment by BrutalBirdie** [comment]
Check the Ansible Lint and fix please.
Closed for now due to inactivity.
Send a message to reopen this PR.

**Pull request closed without merging**